sdk/core/azure-core/src/main/java/com/azure/core/http/policy/AddHeadersFromContextPolicy.java

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import com.azure.core.http.HttpHeader;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.Context;
+import com.azure.core.http.HttpRequest;
+import reactor.core.publisher.Mono;
+
+import static com.azure.core.util.Context.AZURE_REQUEST_HTTP_HEADERS_KEY;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * The pipeline policy that override or add  {@link HttpHeaders} in {@link HttpRequest} by reading values from
+ * {@link Context} with key 'azure-http-headers-key'. The value for this key should be of type
+ * {@link HttpHeaders} for it to be added in {@link HttpRequest}.
+ */
```

Comment:
Include code samples to demonstrate how this would work from an end-users perspective. They don't need to be code snippets because you won't be able to reference the  relevant code in app config or elsewhere, so you will need to add a checkstyle suppression.

Reply:
included sample. Also discussed with team members here about this sample.

Reply:
@JonathanGiles   Please look into the code sample. This would be last thing before I merge PR. Need your approval.